subprojects/core/src/main/java/org/gradle/caching/internal/tasks/BuildCacheKeyInputs.java

```diff
@@ -37,7 +36,14 @@
     private final ImmutableSortedMap<String, HashCode> inputHashes;
     private final ImmutableSortedSet<String> outputPropertyNames;
 
-    public BuildCacheKeyInputs(String taskClass, HashCode classLoaderHash, List<HashCode> actionClassLoaderHashes, ImmutableList<String> actionClassNames, ImmutableSortedMap<String, HashCode> inputHashes, ImmutableSortedSet<String> outputPropertyNames) {
+    public BuildCacheKeyInputs(
+        @Nullable String taskClass,
```

Comment:
I see how `classLoaderHash` can be null, but the other properties should never be null.